runtime/dancebox/src/lib.rs

```diff
@@ -704,13 +724,65 @@ impl pallet_authority_assignment::Config for Runtime {
     type AuthorityId = NimbusId;
 }
 
+pub const FIXED_BLOCK_PRODUCTION_COST: u128 = 1 * currency::MICRODANCE;
+
+pub struct BlockProductionCost<Runtime>(PhantomData<Runtime>);
+impl ProvideBlockProductionCost<Runtime> for BlockProductionCost<Runtime> {
+    fn block_cost(_para_id: &ParaId) -> (u128, Weight) {
+        (FIXED_BLOCK_PRODUCTION_COST, Weight::zero())
+    }
+}
+
+parameter_types! {
+    // 60 days worth of blocks
+    pub const MaxCreditsStored: BlockNumber = 60 * 7200;
+}
+
+impl pallet_services_payment::Config for Runtime {
+    type RuntimeEvent = RuntimeEvent;
+    /// Handler for fees
+    type OnChargeForBlockCredit = ChargeForBlockCredit<Runtime>;
+    /// Currency type for fee payment
+    type Currency = Balances;
+    /// Provider of a block cost which can adjust from block to block
+    type ProvideBlockProductionCost = BlockProductionCost<Runtime>;
+    /// The maximum number of credits that can be accumulated
+    type MaxCreditsStored = MaxCreditsStored;
+    type WeightInfo = pallet_services_payment::weights::SubstrateWeight<Runtime>;
+}
+
+pub struct InflationRewardsAndServicesPayment;
+
+impl AuthorNotingHook<AccountId> for InflationRewardsAndServicesPayment {
+    fn on_container_author_noted(
+        author: &AccountId,
+        block_number: BlockNumber,
+        para_id: ParaId,
+    ) -> Weight {
+        let mut weight = Weight::zero();
+
+        weight = weight.saturating_add(InflationRewards::on_container_author_noted(
```

Comment:
Nice, now that we are clearly adding up on_containter_author_noted, maybe this could be implemented for tuples and pass (`InflationRewards`, `ServicesPayment`)

Reply:
I thought about that, but this way will be easier to maintain, in case we want some additional logic such as only executing a hook if the other one succeeded.

Unless there's some macro that implements a trait for tuples and will make this more readable.

Reply:
there is this one yes: https://github.com/paritytech/polkadot-sdk/blob/29654a4d7154296e8e6cb2a067896490d67d96a1/polkadot/xcm/xcm-builder/src/fee_handling.rs#L39

Reply:
We will need to see wether it goes one by one or wether it returns in the first successful one

Reply:
I think this one is close to what you want: https://github.com/paritytech/polkadot-sdk/blob/29654a4d7154296e8e6cb2a067896490d67d96a1/polkadot/runtime/parachains/src/paras/mod.rs#L500